lib/Basic/LangOptions.cpp

```diff
@@ -146,9 +157,10 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
   bool UnsupportedOS = false;
 
   // Set the "os" platform condition.
-  if (Target.isMacOSX())
+  if (Target.isMacOSX()) {
     addPlatformConditionValue(PlatformConditionKind::OS, "OSX");
-  else if (triple.isTvOS())
+    addPlatformConditionValue(PlatformConditionKind::OS, "macOS");
```

Comment:
I'm uncomfortable with this because I don't think this is a distinction worth preserving anywhere but when doing source things. I'd much rather do the canonicalization in ParseIfConfig.cpp than have all platform conditionals support this.

Reply:
Are you fine with keeping this alias in "checkPlatformConditionAndValue" function? It seems to be a better way than tweaking the AST itself.

Reply:
Hm, all right. It's not like we're ever going to be able to take out the alias anyway.

Reply:
Nitpick: I would probably just name it `checkPlatformCondition`.